### PR TITLE
Updating the AWS Prompt_Engineering Notebook + Adding an Example of ReAct with Llama 2 on Bedrock

### PR DESCRIPTION
**Pull request by ejsd1989**
# What does this PR do?

Prompt_Engineering_with_Llama_2_On_Amazon_Bedrock.ipynb
+ Added in a section to help better understand how to use the [INST] tags, updated some examples, and added reference to the Deeplearning.AI Prompt Engineering course

ReAct_Llama_2_Bedrock-WK.ipynb
+ Adding an example of ReAct using Llama 2
+ Short explanation of ReAct
+ Setup and configuration to use Amazon Bedrock
+ Example of using the Bedrock api via langchain
+ Setup for use of DuckDuckGoSearchRun, WikipediaAPIWrapper, and PythonREPL.
+ Created a pattern for the model to follow in order to use the tools and do reasoning similar to CoT.
+ Cleaned up and formatted the generated text before giving it to the corresponding tool.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/llama-recipes/blob/main/CONTRIBUTING.md#pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes?  Documentation is in the notebooks
- [ ] Did you write any new necessary tests?

Thanks for contributing 🎉!

### CODE REVIEW

File: examples/examples_with_aws/Prompt_Engineering_with_Llama_2_On_Amazon_Bedrock.ipynb [hunk 1]
```diff
@@ -362,21 +384,74 @@
...
+    "print(response)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### [INST] Prompt Tags\n",
+    "\n",
+    "To signify user instruction to the Model, you may use the `[INST][/INST]` tags, and the model response will filter have the tags filtered out. The tags help to signify that the enclosed text are instructions for the model to follow and use in the response.\n",
```
Comment (jeffxtang) [inline examples/examples_with_aws/Prompt_Engineering_with_Llama_2_On_Amazon_Bedrock.ipynb:408]: You may want to add a note that some LLM integration code with LangChain already takes care of adding [INST] and [/INST] around the user prompt so the app developer won't need to manually add it. (I'm not sure about the case with Bedrock, but Together needs manually wrapping the [INST] tag while Anyscale doesn't). But under the hood the [INST] is needed for Llama 2 (maybe add the link [here](https://huggingface.co/blog/llama2#how-to-prompt-llama-2) for more info).

File: examples/examples_with_aws/Prompt_Engineering_with_Llama_2_On_Amazon_Bedrock.ipynb
Comment (jeffxtang) [inline examples/examples_with_aws/Prompt_Engineering_with_Llama_2_On_Amazon_Bedrock.ipynb:1]: Under "Note about LangChain", there is `~/.aws/` and `/.aws/config` - can we make them the same (`~/.aws`)?

File: examples/examples_with_aws/ReAct_Llama_2_Bedrock-WK.ipynb [hunk 2]
```diff
@@ -0,0 +1,572 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Advanced Techniques\n",
+    "## 1. ReAct\n",
+    "\n",
+    "LLMs abilities for reasoning (e.g. chain-of-thought CoT prompting) and acting have primarily been studied as separate topics. **ReAct** [Shunyu Yao et al. ICLR 2023](https://arxiv.org/pdf/2210.03629.pdf) (Reason and Act) is a method to generate both reasoning traces and task-specific actions in an interleaved manner.\n",
+    "\n",
+    "In simple words, we define specific patterns for the language model to follow. This allows the model to act (usually through tools) and reason. Hence the model create a squence of interleaved thoughts and actions. Such systems that act on an enviroment are usually called **agents** (borrowed from reinforcement learning).\n",
```
Comment (jeffxtang) [inline examples/examples_with_aws/ReAct_Llama_2_Bedrock-WK.ipynb:12]: typo: "the model create" => "the model creates"

File: examples/examples_with_aws/ReAct_Llama_2_Bedrock-WK.ipynb [hunk 3]
```diff
@@ -0,0 +1,572 @@
...
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "LLAMA2_70B_CHAT = \"meta.llama2-70b-chat-v1\"\n",
+    "LLAMA2_13B_CHAT = \"meta.llama2-13b-chat-v1\"\n",
+    "\n",
+    "# We'll default to the smaller 13B model for speed; change to LLAMA2_70B_CHAT for more advanced (but slower) generations\n",
+    "DEFAULT_MODEL = LLAMA2_70B_CHAT\n",
```
Comment (jeffxtang) [inline examples/examples_with_aws/ReAct_Llama_2_Bedrock-WK.ipynb:90]: typo - default to 70b? or `DEFAULT_MODEL = LLAMA2_13B_CHAT`?

File: examples/examples_with_aws/ReAct_Llama_2_Bedrock-WK.ipynb [hunk 4]
```diff
@@ -0,0 +1,572 @@
...
+      "- Action: duckduckgo_tool\n",
+      "- Action Input: \"Nvidia market cap\"\n",
+      "- Observation: Earnings Yield. 1.51%. PEGY Ratio. 0.1128. Operating PE Ratio. 59.62. Normalized PE Ratio. 66.07. In depth view into NVIDIA Market Cap including historical data from 1999, charts and stats. NVIDIA Market Cap. NVIDIA has a market cap or net worth of $1.98 trillion as of February 28, 2024. Its market cap has increased by 275.81% in one year\n",
+      "- Thought:\n"
+     ]
+    }
+   ],
+   "source": [
+    "response_observation = next_step(response)\n",
+    "new_query = query + '\\033[32m\\033[1m' + response_observation\n",
```
Comment (jeffxtang) [inline examples/examples_with_aws/ReAct_Llama_2_Bedrock-WK.ipynb:454]: maybe add a quick note saying '\\033[32m\\033[1m' contains escape code to set the text that follows to be bold green (another one later to be bold blue.

**Review by jeffxtang — COMMENTED** [review]
@ejsd1989 PR looks good! Just a few minor comments.

File: examples/examples_with_aws/Prompt_Engineering_with_Llama_2_On_Amazon_Bedrock.ipynb
Comment (ejsd1989) [reply examples/examples_with_aws/Prompt_Engineering_with_Llama_2_On_Amazon_Bedrock.ipynb:1]: Updated

**Review by jeffxtang — APPROVED** [review]

**Pull request merged**